src/mesh/coordinates.cxx

```diff
@@ -28,23 +28,30 @@ Coordinates::Coordinates(Mesh *mesh)
       G3_23(mesh), G1(mesh), G2(mesh), G3(mesh), ShiftTorsion(mesh),
       IntShiftTorsion(mesh), localmesh(mesh), location(CELL_CENTRE) {
 
-  if (mesh->get(dx, "dx")) {
+  // If the source does not include y-boundary guard cells then communications
+  // are needed to fill them. This provides backwards compatibility even though
+  // the communicated guard cell values may not be correct, because the
+  // integrated shear is not necessarily continuous when poloidal angle goes
+  // from 2pi->0
+  bool need_comms = !mesh->sourceHasYGuards();
+
+  if (localmesh->get(dx, "dx", 0., need_comms)) {
```

Comment:
think i'm missing your point... need_comms is a local variable, the functionality is in `sourceHasYGuards()` which is a member of Mesh.

Reply:
If `sourceHasGuards` is all that is required then couldn't we put the call to this inside `mesh::get` so that `mesh::get` does the right thing?

Reply:
I was originally going to load zShift in Coordinates (instead of only in ShiftedMetric), and pass 'false' for that, but I think there might be a better solution, and in any case is better to add the extra argument only when it's actually needed, so I'll put the call inside Mesh::get as you suggest.

Reply:
This should all only be done once (or a few times) during initialisation so even if we pay a few extra communications that aren't strictly needed (as we overwrite guards etc.) then I think that is ok if it helps make the code cleaner (e.g. not having to repeat the default argument everywhere).